chunkset.c

```diff
@@ -5,8 +5,10 @@
 #include "zbuild.h"
 #include "zutil.h"
 
-#ifdef UNALIGNED_OK
+#if defined(UNALIGNED64_OK)
 typedef uint64_t chunk_t;
+#elif defined(UNALIGNED_OK)
+typedef uint32_t chunk_t[2];
 #else
 typedef uint8_t chunk_t[8];
 #endif
```

Comment:
This is completely wrong... size of chunk_t is always correct, but the type in function prototype/parameter list is not... Mixing arrays with non-arrays in typedef makes the code confusing and prone to errors.

We don't want uint32_t*[2] as the function parameter type. we want uint32_t* when UNALIGNED64_OK not defined.
We don't want uint8_t*[8] as the function parameter type. we want uint8_t* when neither UNALIGNED_OK or UNALIGNED64_OK is defined..